frontend/src/i18n/request.ts

```diff
@@ -10,7 +10,16 @@ import { getMessagesWithFallbacks } from "./getMessagesWithFallbacks";
  */
 
 // @ts-expect-error TS2345: Argument of type error is expected behavior by next-intl maintainer: https://github.com/amannn/next-intl/issues/991#issuecomment-2050087509
-export default getRequestConfig(async ({ locale }) => {
+export default getRequestConfig(async ({ requestLocale }) => {
+  let locale = (await requestLocale) || "";
+
+  const isValidLocale = locales.includes(locale); // https://github.com/microsoft/TypeScript/issues/26255
```

Comment:
moved this over from the `getMessagesWithFallbacks` code. Not a necessary change but just made more sense to me to do the locale default logic as high up in the call stack as possible